addons/mail/models/mail_message.py

```diff
@@ -147,6 +147,7 @@ def default_get(self, fields):
     # user interface
     starred_partner_ids = fields.Many2many(
         'res.partner', 'mail_message_res_partner_starred_rel', string='Favorited By')
+    pinned_at = fields.Datetime('Pinned', help='Datetime at which the message has been pinned')
```

Comment:
probably missing a partial index (when value is set), otherwise this might become really slow in big channels to find pinned messages?